data/input_files/tests/nonlinear_solid/serac_component_bc/qs_attribute_solve.lua

```diff
@@ -0,0 +1,55 @@
+-- Comparison information
+expected_l2norm = 0.03330115
+epsilon = 0.0001
+
+-- Simulation time parameters
+dt      = 1.0
+
+main_mesh = {
+    -- mesh file
+    mesh = "../../../../meshes/square.mesh",
+    -- serial and parallel refinement levels
+    ser_ref_levels = 2,
+    par_ref_levels = 0,
+}
+
+-- Solver parameters
+nonlinear_solid = {
+    solver = {
+        linear = {
+            rel_tol     = 1.0e-8,
+            abs_tol     = 1.0e-12,
+            max_iter    = 5000,
+            print_level = 0,
+            solver_type = "minres",
+            prec_type   = "L1JacobiSmoother",
+        },
+
+        nonlinear = {
+            rel_tol     = 1.0e-6,
+            abs_tol     = 1.0e-8,
+            max_iter    = 5000,
+            print_level = 1,
+        },
+    },
+
+    -- polynomial interpolation order
+    order = 2,
+
+    -- neo-Hookean material parameters
+    mu = 0.25,
+    K  = 10.0,
+
+    -- boundary condition parameters
+    boundary_conds = {
+        {
+            name = "displacement_x",
```

Comment:
Do we want separate boundary condition names instead of a set of active component directions? I lean toward the latter, but I could be persuaded. Thoughts @samuelpmishLLNL @solberg2 ?

Reply:
That's a good question. I think for certain b.c.'s "tangent" and "normal" are also good descriptions vs `x-,y-,z-`.

Reply:
"Tangent" is a pretty difficult thing to handle in 3D, because of course there are 2 orthogonal tangent directions, and they have to be defined somehow.

For many verification problems simple component directions are good.  What we do is allow the user to specify a set of local component directions "X_LOCAL", "Y_LOCAL", which then we provide a facility for defining these.

Reply:
I think this is good for now, but we probably want to have a larger design discussion about this. I'll create an issue.